app/src/main/java/org/p2p/wallet/infrastructure/network/provider/SeedPhraseProvider.kt

```diff
@@ -31,13 +51,44 @@ class SeedPhraseProvider(
             throw e
         }
 
+    private fun getSeedPhraseProviderFromStorage(): SeedPhraseProviderType? =
+        try {
+            val result: String? = secureStorage.getString(Key.KEY_SEED_PHRASE_PROVIDER)
+            SeedPhraseProviderType.getValueOf(result)
+        } catch (e: Throwable) {
+            Timber.tag(TAG).e(e)
+            throw e
+        }
+
     private fun saveSeedPhraseToStorage(value: List<String>) {
         runBlocking {
             secureStorage.saveObjectList(Key.KEY_SEED_PHRASE, value)
         }
     }
 
+    private fun saveSeedPhraseProviderToStorage(value: SeedPhraseProviderType) {
+        runBlocking {
+            secureStorage.saveString(Key.KEY_SEED_PHRASE_PROVIDER, value.title)
+        }
+    }
+
     fun clear() {
         seedPhrase = emptyList()
     }
 }
+
+enum class SeedPhraseProviderType(val title: String) {
+    MANUAL("Manual"),
+    WEB_AUTH("Web3Auth");
+
+    companion object {
+        fun getValueOf(value: String?) = values().firstOrNull {
+            it.title == value
+        }
+    }
+}
+
+data class UserSeedPhrase(
+    val seedPhrase: List<String>,
+    val provider: SeedPhraseProviderType?
```

Comment:
instead of null you can add new enum entry like "NOT_PROVIDED" or smth like that